### PR TITLE
Lowercase headernames for HTTP/2 and HTTP/3

### PR DESCRIPTION
**Pull request by bazzadp**
Fixes https://github.com/curl/curl/issues/4400

### CODE REVIEW

File: lib/http2.c [hunk 1]
```diff
@@ -2026,6 +2026,8 @@ static ssize_t http2_send(struct connectdata *conn, int sockindex,
       nva[i].namelen = strlen((char *)nva[i].name);
     }
     else {
+      /* Lower case the header name for HTTP/2 */
+      Curl_strntolower((char *)hdbuf, hdbuf, strlen(hdbuf));
```
Comment (bagder) [inline lib/http2.c]: This should probably instead be done after the `namelen` assignment and use that length. Avoids strlen() in vain and uses a better length. I *think* this is why test 1700 triggered valgrind.
~~~c
nva[i].namelen = (size_t)(end - hdbuf);
Curl_strntolower((char *)hdbuf, hdbuf, nva[i].namelen);

File: lib/vquic/ngtcp2.c [hunk 2]
```diff
@@ -1201,6 +1201,8 @@ static CURLcode http_request(struct connectdata *conn, const void *mem,
       nva[i].namelen = strlen((char *)nva[i].name);
     }
     else {
+      /* Lower case the header name for HTTP/3 */
+      Curl_strntolower((char *)hdbuf, hdbuf, strlen(hdbuf));
```
Comment (bagder) [inline lib/vquic/ngtcp2.c]: Same argument here

File: lib/vquic/quiche.c [hunk 3]
```diff
@@ -646,6 +646,8 @@ static CURLcode http_request(struct connectdata *conn, const void *mem,
       nva[i].name_len = strlen((char *)nva[i].name);
     }
     else {
+      /* Lower case the header name for HTTP/3 */
+      Curl_strntolower((char *)hdbuf, hdbuf, strlen(hdbuf));
```
Comment (bagder) [inline lib/vquic/quiche.c]: and here

File: lib/http2.c (hunk 1 above)
Comment (bazzadp) [reply lib/http2.c]: Fixed

File: lib/vquic/ngtcp2.c (hunk 2 above)
Comment (bazzadp) [reply lib/vquic/ngtcp2.c]: Fixed

File: lib/vquic/quiche.c (hunk 3 above)
Comment (bazzadp) [reply lib/vquic/quiche.c]: Fixed

**Comment by bazzadp** [comment]
@bagder , any ideas what to do with this? Looks like timeouts and other CI infrastructure issues rather than necessarily anything wrong with my code (I hope!). Can we rerun those or do I need to submit a change for this to happen?

**Comment by bagder** [comment]
What a sad look. They're all unrelated to your change though.

I think its still fine. I retriggered the Linux ones as they *should* be able to complete. The mac ones that failed are worked on in #4403 so they will be red here.

**Comment by bagder** [comment]
Thanks!

File: lib/strcase.c [hunk 4]
```diff
@@ -93,6 +93,75 @@ char Curl_raw_toupper(char in)
...
+  case 'T':
+    return 't';
+  case 'U':
+    return 'u';
+  case 'V':
+    return 'v';
+  case 'W':
+    return 'w';
+  case 'X':
+    return 'X';
```
Comment (LPardue) [inline lib/strcase.c]: BUG!!!

**Pull request closed without merging**